realm/src/main/java/io/realm/RealmQuery.java

```diff
@@ -79,14 +78,10 @@ public RealmQuery(RealmResults realmList, Class<E> clazz) {
         this.realm = realm;
         this.clazz = clazz;
         this.query = query;
-        TableOrView dataStore = getTable();
+        this.tableSpec = realm.getTable(clazz);
         this.columns = Realm.columnIndices.get(clazz.getSimpleName());
     }
 
-    TableOrView getTable() {
-        return realm.getTable(clazz);
-    }
-
     private boolean containsDot(String s) {
```

Comment:
Not related to the PR but I wonder why this is here. And why not using s.contains('.'). The compiler should do all the needed optimizations